src/key.rs

```diff
@@ -113,7 +150,16 @@ impl SecretKey {
         SecretKey(data)
     }
 
-    /// Converts a `SECRET_KEY_SIZE`-byte slice to a secret key
+    /// Converts a `SECRET_KEY_SIZE`-byte slice to a secret key.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(feature="rand")] {
+    /// use secp256k1::{rand, SecretKey};
```

Comment:
rand isn't really used here

Reply:
Oh cheers, I've not worked out how to lint the examples yet.

Reply:
I haven't tried it but I'd expect `--all-targets` on `cargo clippy` to include rustdoc.

Reply:
I only learned during this PR that `-all-targets` does _not_ include docs (I have my editor configured to run clippy with `--all-targets`, thanks Lucas :)